.github/workflows/run_tests.yml

```diff
@@ -85,6 +85,9 @@ jobs:
           # Use "eager" update strategy in case cached dependencies are outdated
           # Using regular install NOT editable install: see GH #3020
           pip install --upgrade --upgrade-strategy eager '.[${{ matrix.extras }},plots]'
+      - name: Run mypy
+        if: matrix.extras == 'test-core'
+        run: mypy shap
```

Comment:
Would this make more sense as a separate `job` that runs in parallel?

Reply:
absolutely, will do. Would split the `test-core` job and add mypy to that. 

Note: Will check test coverage for the plotting functions I changed, to make sure that I didn't break anything. Afterwards I'll either add more tests or would consider this as ready. Just forgot to create this as a draft PR (and don't know if I can promote it back to "draft PR")

Reply:
I think conceptually the "test-core" job belongs with the unit tests, as it requires the other common steps like uploading code coverage and running pytest.

May I suggest leaving that unit test job with the others in that matrix, and making the new job just do type checking?

Reply:
I think the new job could be pretty much as simple as `pip install mypy` , `mypy shap`. It doesn't need caching or unit tests

Reply:
done, would you be so kind and take a final look?